Find a functionally equivalent code:android.os.Parcel.writeArrayMapInternal:COMMENT
Method Modifier: 
Comment:/* package */

@@ -10,19 +10,22 @@
         here.fillInStackTrace();
         Log.d(TAG, "Writing " + N + " ArrayMap entries", here);
     }
+    int startPos;
     for (int i = 0; i < N; i++) {
         if (DEBUG_ARRAY_MAP)
-            Log.d(TAG, "  Write #" + i + ": key=0x" + (val.keyAt(i) != null ? val.keyAt(i).hashCode() : 0) + " " + val.keyAt(i));
-        writeValue(val.keyAt(i));
+            startPos = dataPosition();
+        writeString(val.keyAt(i));
         writeValue(val.valueAt(i));
+        if (DEBUG_ARRAY_MAP)
+            Log.d(TAG, "  Write #" + i + " " + (dataPosition() - startPos) + " bytes: key=0x" + Integer.toHexString(val.keyAt(i) != null ? val.keyAt(i).hashCode() : 0) + " " + val.keyAt(i));
     }
 }

